github/github.go

```diff
@@ -554,7 +554,23 @@ func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) (*Res
 
 		return nil, err
 	}
-	defer resp.Body.Close()
+
+	defer func() {
+		// Ensure the response body is fully read and closed
+		// before we reconnect, so that we reuse the same TCP connection.
+		// Close the previous response's body. But read at least some of
+		// the body so if it's small the underlying TCP connection will be
+		// re-used. No need to check for errors: if it fails, the Transport
+		// won't reuse it anyway.
+		const maxBodySlurpSize = 2 << 10
+		if resp.ContentLength == -1 || resp.ContentLength <= maxBodySlurpSize {
+			io.CopyN(ioutil.Discard, resp.Body, maxBodySlurpSize)
+		}
+
+		if rerr := resp.Body.Close(); err == nil {
```

Comment:
I think, this if condition can be removed because `err` is not part of the named values. Therefore, the error is never returned if any occur.

Reply:
@philippfranke Thank you for the review. I am glad that you liked the PR.
Yes, I also agree with your point. There is no need to check for error as it is not part of the named return values. Also, If it fails, the Transport won't reuse it anyway.

Reply:
@philippfranke I just changed the code. Please let me know if it seems good to you. :)